File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.java
class Solution {
   
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer,Integer> count = new HashMap<>();
        for(int i:nums){
            count.put(i,count.getOrDefault(i,0)+1);
        }
        int n=nums.length;
        int[] ans = new int[k];
        List<Integer>[] fre = new List[n+1];
        for(int i=0;i<fre.length;i++){
            fre[i]=new ArrayList<Integer>();
        }
        for(Map.Entry<Integer,Integer> a : count.entrySet()){
            fre[a.getValue()].add(a.getKey());
        }
        int index=0;
        for(int i=n;i>0 && index<k;i--){
            for(int num:fre[i]){
                ans[index]=num;
                index++;
                if(index==k)return ans;
            }          
        }
        return ans;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        nums = sorted(nums)
        ans = set()
        for i in range(n):
            start = i+1
            end = n-1
            while start < end:
                if nums[start] + nums[end] + nums[i] == 0:
                    ans.add((nums[i], nums[start], nums[end]))
                    start += 1
                    end -= 1
                elif nums[start] + nums[end] + nums[i] > 0:
                    end -= 1
                else:
                    start += 1
        l = []

        for i in ans:
            l.append(list(i))
        return l
